src/Network/Network/NetworkWatcher/ConnectionMonitor/ConnectionMonitorBaseCmdlet.cs

```diff
@@ -88,5 +109,637 @@ public MNM.NetworkWatcher GetNetworkWatcherByLocation(string location)
+                MonitoringStatus = ConnectionMonitor.MonitoringStatus,
+                Tags = NetworkResourceManagerProfile.Mapper.Map<Dictionary<string, string>>(ConnectionMonitor.Tags),
+                ConnectionMonitorType = ConnectionMonitor.ConnectionMonitorType,
+                Source = NetworkResourceManagerProfile.Mapper.Map<PSConnectionMonitorSource>(ConnectionMonitor.Source),
+                Destination = NetworkResourceManagerProfile.Mapper.Map<PSConnectionMonitorDestination>(ConnectionMonitor.Destination),
+            };
+
+            return psConnectionMonitor;
+        }
+
+        public MNM.ConnectionMonitorTestConfiguration GetTestConfigurationByName(IList<ConnectionMonitorTestConfiguration> TestConfigurations, string TestConfigName)
+        {
+            if (TestConfigurations == null || string.IsNullOrEmpty(TestConfigName))
+                return null;
+
+            foreach (ConnectionMonitorTestConfiguration TestConfiguration in TestConfigurations)
+            {
+                if (string.Compare(TestConfigName, TestConfiguration.Name) == 0)
+                {
+                    return TestConfiguration;
+                }
+            }
+
+            return null;
+        }
+
+        public MNM.ConnectionMonitorEndpoint GetEndpoinByName(IList<ConnectionMonitorEndpoint> Endpoints, string EndpointName)
+        {
+            if (Endpoints == null || string.IsNullOrEmpty(EndpointName))
+                return null;
+
+            foreach (ConnectionMonitorEndpoint Endpoint in Endpoints)
+            {
+                if (string.Compare(EndpointName, Endpoint.Name) == 0)
+                {
+                    return Endpoint;
+                }
+            }
+
+            return null;
+        }
+
+    public bool ValidateConnectionMonitorV1V2Parameters(string SourceResourceId, string DestinationResourceId, PSConnectionMonitorResult InputObject, string DestinationAddress, int? MonitoringIntervalInSeconds, List<PSNetworkWatcherConnectionMonitorTestGroupObject> TestGroups, List<PSNetworkWatcherConnectionMonitorOutputObject> Outputs)
+        {
+            if ((!string.IsNullOrEmpty(SourceResourceId)     || 
+                 !string.IsNullOrEmpty(DestinationResourceId)||
+                 !string.IsNullOrEmpty(DestinationAddress)   ||
+                 MonitoringIntervalInSeconds != null) && (TestGroups != null || InputObject !=null))
+            {
+                throw new ArgumentException("Connection moniotr V1 can not be defined with either TestGroup. Either connection monitor V1 or V2 can be specified");
+            }
+
+            if (string.IsNullOrEmpty(SourceResourceId) && InputObject == null && TestGroups == null)
+            {
+                throw new ArgumentException("Either SourceResourceId or InputObject or TestGroups is to be ddefined");
+            }
+
+            // Validate Test Group
+            if (TestGroups != null && TestGroups.Any())
```

Comment:
TestGroups.Any() covers both cases.

Reply:
I though it will gives expcetion if TestGroups is null?

Reply:
it test for testGroups is not null first and if so, run .any. otherwise, use .any only will gives expcetion if test group is null